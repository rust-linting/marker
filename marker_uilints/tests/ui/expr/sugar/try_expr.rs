#![feature(register_tool)]
#![register_tool(marker)]

fn try_something() -> Option<u32> {
    #[warn(marker::print_every_expr)]
    Some(21)?;
    None
}

fn kanske_option() -> Result<i32, u32> {
    let x: Result<i32, u32> = Err(1);
    #[warn(marker::print_every_expr)]
    x?;
    Ok(42)
}

fn main() {}
