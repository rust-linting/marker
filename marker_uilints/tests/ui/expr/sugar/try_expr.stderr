warning: expr
 --> $DIR/try_expr.rs:3:5
  |
3 |     Some(21)?;
  |     ^^^^^^^^^
  |
  = note: SpanSource: File(
              FileInfo {
                  file: "$DIR/try_expr.rs",
                  span_src: SpanSrcId(..),
              },
          )
  = note: Snippet: "Some(21)?"
note: the lint level is defined here
 --> $DIR/try_expr.rs:2:12
  |
2 |     #[warn(marker::marker_uilints::print_every_expr)]
  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: expr
 --> $DIR/try_expr.rs:3:5
  |
3 |     Some(21)?;
  |     ^^^^^^^^
  |
  = note: SpanSource: File(
              FileInfo {
                  file: "$DIR/try_expr.rs",
                  span_src: SpanSrcId(..),
              },
          )
  = note: Snippet: "Some(21)"

warning: expr
 --> $DIR/try_expr.rs:3:10
  |
3 |     Some(21)?;
  |          ^^
  |
  = note: SpanSource: File(
              FileInfo {
                  file: "$DIR/try_expr.rs",
                  span_src: SpanSrcId(..),
              },
          )
  = note: Snippet: "21"

warning: expr
  --> $DIR/try_expr.rs:10:5
   |
10 |     x?;
   |     ^^
   |
   = note: SpanSource: File(
               FileInfo {
                   file: "$DIR/try_expr.rs",
                   span_src: SpanSrcId(..),
               },
           )
   = note: Snippet: "x?"
note: the lint level is defined here
  --> $DIR/try_expr.rs:9:12
   |
9  |     #[warn(marker::marker_uilints::print_every_expr)]
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: expr
  --> $DIR/try_expr.rs:10:5
   |
10 |     x?;
   |     ^
   |
   = note: SpanSource: File(
               FileInfo {
                   file: "$DIR/try_expr.rs",
                   span_src: SpanSrcId(..),
               },
           )
   = note: Snippet: "x"

warning: 5 warnings emitted

